orchagent/portsorch.cpp

```diff
@@ -2524,6 +2554,18 @@ bool PortsOrch::setGearboxPortAttr(const Port &port, dest_port_type_t port_type,
                     m_gearboxTable->hset(key, speed_attr, to_string(speed));
                     SWSS_LOG_NOTICE("BOX: Updated APPL_DB key:%s %s %d", key.c_str(), speed_attr.c_str(), speed);
                 }
+                else if (id == SAI_PORT_ATTR_FEC_MODE && fec_override_sup)
+                {
+                    attr.id = SAI_PORT_ATTR_AUTO_NEG_FEC_MODE_OVERRIDE;
+                    attr.value.booldata = true;
+
+                    status = sai_port_api->set_port_attribute(dest_port_id, &attr);
```

Comment:
@prgeor Both are independent attributes. SAI headers doesn't mandate that FEC override should need to be set only on AN. To avoid complexity in orchagent, both are set independently and vendor SAI will handle them in any order

Reply:
@dgsudharsan please check the defintion of the attribute here https://github.com/opencomputeproject/SAI/blame/master/inc/saiport.h#L1913
```
 * If set to true, any auto-negotiated FEC mode will be
     * overridden by the value configured in SAI_PORT_ATTR_FEC_MODE
```